app/src/main/java/org/yuttadhammo/BodhiTimer/TimerActivity.kt

```diff
@@ -108,6 +105,35 @@ class TimerActivity : AppCompatActivity(), View.OnClickListener, OnSharedPrefere
         createNotificationChannel(context!!)
     }
 
+    override fun onNewIntent(intent: Intent?) {
+        super.onNewIntent(intent)
+        checkForDeepLink(intent)
+    }
+
+    private fun checkForDeepLink(intent: Intent?) {
+        val data: Uri = intent?.data ?: return
+
+        val paramNames = data.getQueryParameterNames()
+
+        if (paramNames.contains("times")) {
+            val tL = TimerList()
+
+            var notificationUri = prefs.getString("NotificationUri", Sounds.DEFAULT_SOUND)
+
+            when (notificationUri) {
+                "system" -> notificationUri = prefs.getString("SystemUri", "")
+                "file" -> notificationUri = prefs.getString("FileUri", "")
+            }
+
+            for (timeStr in data.getQueryParameter("times")!!.split(",").toTypedArray()) {
+                val time = Integer.valueOf(timeStr)
```

Comment:
Have you tested what happens
* when ``times=`` (empty param)
* when ``times=,``
* when ``times=@€(@,€(``

Reply:
No, good point... The last two raise an unhandled error. I've added a try/catch and a check whether the timer list is empty after parsing.